core/src/main/java/org/ldaptive/auth/SearchDnResolver.java

```diff
@@ -262,7 +287,21 @@ public String resolve(final User user)
    */
   protected String resolveDn(final LdapEntry entry)
   {
-    return entry.getDn();
+    String dn = entry.getDn();
+
+    if (resolveFromAttribute != null) {
+      LdapAttribute attr = entry.getAttribute(resolveFromAttribute);
+
+      if (attr.size() != 1) {
+        logger.info("Skipping candidate as it does not meet cardinality (must contain a single value), in dn: " + dn + " resolveDnFromAttribute: " + resolveFromAttribute);
+      } else if(attr.isBinary()) {
+        logger.info("Skipping candidate as it is binary, in dn: " + dn + " resolveDnFromAttribute: " + resolveFromAttribute);
```

Comment:
Change these `info` statements to `warn`. Also, use parameterized logging.

Reply:
I agree in both and implemented accordingly.